src/main/python/vantiqsdk.py

```diff
@@ -0,0 +1,1399 @@
+                             'Missing content type specification for an upload call().',
+                             [])
+        elif resource is None:
+            ve = VantiqError('io.vantiq.python.upload.resource',
+                             'The resource type is missing for an upload() call.',
+                             [])
+        elif filename is None and inmem is None:
+            ve = VantiqError('io.vantiq.python.content',
+                             'The filename and inmem parameters for an upload() call are both missing.',
+                             [])
+        elif filename is not None and doc_name is not None and inmem is not None:
+            ve = VantiqError('io.vantiq.python.duplicatename',
+                             'When uploading in memory object, only one of filename or doc_name is permitted.',
+                             [])
+        elif filename is None and doc_name is None:
+            ve = VantiqError('io.vantiq.python.noname',
+                             'Neither a file name nor a document name was provided for an upload() call.',
+                             [])
+
+        if ve is not None:
+            return VantiqResponse.from_error(ve)
+
+        path = self._build_path(resource, None)
+        headers = {aiohttp.hdrs.AUTHORIZATION: self._auth_header}
+
+        try:
+            resp = await self._connection.upload(path, headers, content_type, filename, doc_name, inmem)
+            ret_val = VantiqResponse(resp.ok, resp.status, resp.content_type)
+            ret_val._populate_count(resp)
+            await ret_val._populate_body(resp)
+            return ret_val
+
+        except VantiqException:
+            # If we've already handled or wrapped it, just pass it along
+            raise
+        except Exception as e:
+            raise VantiqException('io.vantiq.python.operationerror',
+                                  'Unexpected error during {0} operation.',
+                                  ['upload']) from e
+
+    async def count(self, resource: str, where: Union[dict, None]) -> VantiqResponse:
+        """(Async) Return the number of items in a Vantiq resource that satisfy the where clause
+
+        Parameters:
+            resource : str
+                The name of the resource to be counted.  System resource names are provided via the VantiqResource
+                class.
+            where : dict(str: *)
+                (optional) The "where clause" to be used to restrict the counting.  The where clause is defined
+                in the API Reference Guide.
+        Returns:
+            VantiqResponse where the `count` field contains the count requested.
+        """
+
+        operation = 'count'
+        try:
+            query_params = {'count': 'true'}
+            if where is not None:
+                query_params['where'] = json.dumps(where)
+            props = ['_id']  # Since we don't care about the data, return the least we can
```

Comment:
Actually, you can return even less by setting `limit` to `1`.  This will still get you the full count for the query, but only return one actual record.